File: backend/src/main/java/ch/rodano/core/services/bll/configurator/ResourceCategoryGrantsService.java
package ch.rodano.core.services.bll.configurator;

import java.util.List;
import java.util.Map;
import java.util.UUID;

public interface ResourceCategoryGrantsService {

	Map<UUID, List<UUID>> getResourceCategoryGrants(UUID projectId);

	void saveResourceCategoryGrants(UUID projectId, Map<UUID, List<UUID>> profileResourceCategoryMap);
}

File: frontends/main/src/app/configurator/services/api/resource-category-grants.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';

@Injectable({providedIn: 'root'})
export class ResourceCategoryGrantsService {
	constructor(private http: HttpClient) {}

	getResourceCategoryGrants(projectId: string): Observable<Record<string, string[]>> {
		return this.http.get<Record<string, string[]>>(`/api/superuser/configurator/projects/${projectId}/config/category-grants`);
	}

	saveResourceCategoryGrants(projectId: string, grants: Record<string, string[]>): Observable<void> {
		return this.http.put<void>(`/api/superuser/configurator/projects/${projectId}/config/category-grants`, grants);
	}
}
